staging/src/k8s.io/apiserver/pkg/server/config.go

```diff
@@ -911,7 +911,10 @@ func (c completedConfig) New(name string, delegationTarget DelegationTarget) (*G
 	if s.isPostStartHookRegistered(priorityAndFairnessConfigConsumerHookName) {
 	} else if c.FlowControl != nil {
 		err := s.AddPostStartHook(priorityAndFairnessConfigConsumerHookName, func(hookContext PostStartHookContext) error {
-			go c.FlowControl.Run(hookContext.Done())
+			if err := c.FlowControl.Start(hookContext); err != nil {
```

Comment:
this blocks until Start returns, which blocks until informers sync, right? will that make the apiserver stay unhealthy and get restarted?

Reply:
The commented line is in a post-start-hook. The current rev of this PR will make that post-start-hook not return until that Start method returns, which indeed entails those informers syncing their caches. Is there a criterion for how quickly a post-start-hook needs to complete? Is there a way to compare that with how long we expect those informers to take to sync?

Reply:
I don't have those interactions paged in, I would have to check... just pointing out that this PR is not just a logging change, it is a behavior change that will make that specific post-start hook stay unhealthy logger, potentially much longer if some informers fail to sync permanently (like ones for CRDs depending on conversion webhooks which are unhealthy)

Reply:
@sttts: Since the code before this PR is forking the whole "Run" call, I would be OK with forking the "Start" call.

Reply:
Ack. Have changed it.